core/arch/arm/include/kernel/thread.h

```diff
@@ -308,6 +323,9 @@ void thread_restore_irq(void);
 #define THREAD_EXCP_ABT	(1 << 2)
 #define THREAD_EXCP_ALL	(THREAD_EXCP_FIQ | THREAD_EXCP_IRQ | THREAD_EXCP_ABT)
 
+/* Defines the bit mask for the foreign interrupts */
+#define THREAD_FOREIGN_INTR	THREAD_EXCP_IRQ
```

Comment:
> Why are we keeping the old THREAD_EXCP_IRQ and THREAD_EXCP_FIQ?
THREAD_EXCP_IRQ and THREAD_EXCP_FIQ are for the interrupts bit masks of IRQ and FIQ.
They are also used as the bit masks to update the IRQ/FIQ masks of CPU. (e.g. in CPSR, PSTATE).

We defined "native" and "foreign" interrupts are from the "function" point of view. That means "native" is for the interrupts handled by optee_os and "foreign" by "non" optee_os SW (arm-tf, rich OS).
For GICv2
#define THREAD_FOREIGN_INTR	THREAD_EXCP_IRQ
For GICv3 (will introduce in later patches as this patch is just for renaming)
#define THREAD_FOREIGN_INTR	THREAD_EXCP_FIQ

The macro "THREAD_FOREIGN_INTR" is the unique name used for the places that want to mask the "foreign" interrupts (IRQ for GICv2, FIQ for GICv3).